soroban-env-host/src/events/mod.rs

```diff
@@ -23,12 +31,110 @@ pub struct HostEvent {
 #[derive(Clone, Debug)]
 pub enum Event {
     Contract(crate::xdr::ContractEvent),
-    // Debug events are metered and will not be reported in tx meta
-    Debug(DebugEvent),
     // StructuredDebug should not affect metering
     StructuredDebug(crate::xdr::ContractEvent),
 }
 
+fn display_address(addr: &ScAddress, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+    match addr {
+        ScAddress::Account(acct) => match &acct.0 {
+            PublicKeyTypeEd25519(e) => write!(f, "Address(Account({}))", e),
+        },
+        ScAddress::Contract(hash) => write!(f, "Address(Contract({}))", hash),
+    }
+}
+
+fn display_scval(scv: &ScVal, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+    match scv {
+        ScVal::Bool(v) => write!(f, "{}", v),
+        ScVal::Void => write!(f, "Void"),
+        ScVal::Error(e) => write!(f, "Error({},{})", e.type_, e.code),
+        ScVal::U32(v) => write!(f, "{}", v),
+        ScVal::I32(v) => write!(f, "{}", v),
+        ScVal::U64(v) => write!(f, "{}", v),
+        ScVal::I64(v) => write!(f, "{}", v),
+        ScVal::Timepoint(v) => write!(f, "TimePoint({})", v.0),
+        ScVal::Duration(v) => write!(f, "Duration({})", v.0),
+        ScVal::U128(v) => write!(f, "{}", u128::from(v)),
+        ScVal::I128(v) => write!(f, "{}", i128::from(v)),
+        ScVal::U256(v) => write!(
+            f,
+            "{}",
+            u256_from_pieces(v.hi_hi, v.hi_lo, v.lo_hi, v.lo_lo)
+        ),
+        ScVal::I256(v) => write!(
+            f,
+            "{}",
+            i256_from_pieces(v.hi_hi, v.hi_lo, v.lo_hi, v.lo_lo)
+        ),
+        ScVal::Bytes(v) => write!(f, "Bytes({})", v.0),
+        ScVal::String(v) => write!(f, "\"{}\"", v.0),
+        ScVal::Symbol(v) => write!(f, "{}", v.0),
```

Comment:
nit: Shall we do `Symbol({})` format for clarity? Othewise this seems a bit too easy to confuse with string

Reply:
I tried putting some kind of marker on it and it just reads badly. strings are quoted, so in practice it seems not too bad (and this is just for diagnostic display anyway, when the user gets an error dumped out on the console)